Packs/MicrosoftDefenderThreatIntelligence/Integrations/MicrosoftDefenderThreatIntelligence/MicrosoftDefenderThreatIntelligence.py

```diff
@@ -0,0 +1,556 @@
+    client.ms_client.get_access_token()
+    return "✅ Authorization completed successfully."
+
+
+def test_connection(client: Client) -> str:  # pragma: no cover
+    client.ms_client.get_access_token()
+    return "✅ Success!"
+
+
+def article_list_command(client: Client, args: dict[str, Any]) -> CommandResults:
+    article_id = args.get("article_id", "")
+    odata = args.get("odata", "")
+    limit = args.get("limit", 50)
+
+    # Call the Client function and get the raw response
+    response = client.article_list(article_id, odata, limit)
+    display_data = [{"Article Id": article.get("id"), "Title": article.get("title")} for article in response]
+    return CommandResults(
+        "MSGDefenderThreatIntel.Article",
+        "id",
+        outputs=response,
+        readable_output=tableToMarkdown(
+            "Articles:",
+            display_data,
+            removeNull=True,
+        ),
+    )
+
+
+def article_indicators_list_command(client: Client, args: dict[str, Any]) -> CommandResults:
+    article_id = args.get("article_id", "")
+    article_indicator_id = args.get("article_indicator_id", "")
+    odata = args.get("odata", "")
+    limit = args.get("limit", 50)
+    ensure_only_one_argument_provided(article_id=article_id, article_indicator_id=article_indicator_id)
+    response = client.article_indicator_list(article_id, article_indicator_id, odata, limit)
+    display_data = [{"ID": indicator.get("id"), "Artifact Id": indicator.get("artifact", {}).get("id")} for indicator in response]
+    return CommandResults(
+        "MSGDefenderThreatIntel.ArticleIndicator",
+        "id",
+        outputs=response,
+        readable_output=tableToMarkdown(
+            "Indicators:",
+            display_data,
+            removeNull=True,
+        ),
+    )
+
+
+def profile_list_command(client: Client, args: dict[str, Any]) -> CommandResults:
+    intel_profile_id = args.get("intel_profile_id", "")
+    odata = args.get("odata", "")
+    limit = args.get("limit", 50)
+
+    # Call the Client function and get the raw response
+    response = client.profile_list(intel_profile_id, odata, limit)
+
+    display_data = [{"Profile ID": profile.get("id"), "Title": profile.get("title")} for profile in response]
+    return CommandResults(
+        "MSGDefenderThreatIntel.Profile",
+        "id",
+        outputs=response,
+        readable_output=tableToMarkdown(
+            "Profiles:",
+            display_data,
+            removeNull=True,
+        ),
+    )
+
+
+def profile_indicators_list_command(client: Client, args: dict[str, Any]) -> CommandResults:
+    intel_profile_id = args.get("intel_profile_id", "")
+    intel_profile_indicator_id = args.get("intel_profile_indicator_id", "")
+    odata = args.get("odata", "")
+    limit = args.get("limit", 50)
+
+    ensure_only_one_argument_provided(intel_profile_id=intel_profile_id, intel_profile_indicator_id=intel_profile_indicator_id)
```

Comment:
If you can use only one of the following please add it in the .yml description of the arguments.